### ISSUE TITLE
Mechanism for user to "swap out" or "passthrough" an attribute of a class in the lutaml-model class tree

### ISSUE THREAD

**Issue by ronaldtse** [lutaml/lutaml-model#142]
In the oasis-etm gem, the `<entry>` element is meant to accept anything (in XML) that user wishes.

Clearly, this content won't be defined in the oasis-etm gem but by the user.

There are 2 ways the user wants to handle this:
* As passthrough, i.e. treat this content as a raw string.
* Swap the "attribute :content, :string" attribute of the `Entry` class to another Lutaml::Model::Serializable class.

In the first case, we need a way for the user to say that "give me this string as it is without parsing it as XML".

In the second case, we need a way for users of this gem to "swap out" the Entry class with their own "SpecialEntry" so they can process the content of Entry.

Both `map_all` and `raw: true` do not seem to be proper solutions to this use case.

Let's propose a syntax to support both cases..

## Illustration from original issue

In the XML file, it contains this content:

```xml
<row>
  <entry>b1</entry>
  <entry>b2</entry>
  <entry>b3</entry>
  <entry>b4</entry>
  <entry morerows='1' valign='middle'><para>  <!-- Pernicous Mixed Content -->
  Vertical Span</para></entry>
</row>
```

The element `<entry ...><para>...</para></entry>` fails because the `<para>...</para>` content is not defined in lutaml-model, and therefore omitted from the re-generated content.

This is a case where the content of an element is not defined, and we need to implement some pass-through.

Clearly, `map_all` won't work here because we still need to retain the attributes.

So I thought of using [`raw: true`](https://github.com/lutaml/lutaml-model#attribute-as-raw-string):
```ruby
module Oasis
  module Etm
    class Entry < Lutaml::Model::Serializable
      attribute :colname, :string
      attribute :content, :string, raw: true
      # ...

      xml do
        root "entry"

        # Attribute mappings
        map_attribute "colname", to: :colname
        # ...

        # Content mapping
        map_content to: :content
      end
    end
  end
end
```

But it still doesn't work.

_Originally posted by @ronaldtse in https://github.com/lutaml/oasis-etm/issues/1#issuecomment-2451449564_

**Comment by HassanAkbar** [comment]
@ronaldtse By looking at the issue I think 
- We can allow `map_attribute` with `map_all`, With this we can map all the content and map attributes to different values or we can also give a `map_all_attributes` method to map all the attributes to a hash or string.
- We can allow something like `Oasis::Etm::Entry.add_attribute` or `Oasis::Etm::Entry.add_element` and also `Oasis::Etm::Entry.remove_element`, syntax can also be like `Oasis::Etm::Entry.add("entry", type: :element)`

@ronaldtse What do you think about this?

**Comment by ronaldtse** [comment]
> We can allow map_attribute with map_all, With this we can map all the content and map attributes to different values or we can also give a map_all_attributes method to map all the attributes to a hash or string.

Then we should have "map_all", "map_all_content" and "map_all_attributes"?

> We can allow something like Oasis::Etm::Entry.add_attribute or Oasis::Etm::Entry.add_element and also Oasis::Etm::Entry.remove_element, syntax can also be like Oasis::Etm::Entry.add("entry", type: :element)

Let me think...

**Comment by HassanAkbar** [comment]
> Then we should have "map_all", "map_all_content" and "map_all_attributes"?

@ronaldtse Currently `map_all` is doing the same thing that we will be expecting from `map_all_content`. I think we can rename `map_all` to `map_all_content` and add a new method for `map_all_attributes`.

What do you suggest?

**Comment by ronaldtse** [comment]
> we can rename map_all to map_all_content and add a new method for map_all_attributes

Agree. Let's do that.

However, we will still need a way to override a particular attribute/class in a lutaml-model class tree.

**Comment by HassanAkbar** [comment]
> Agree. Let's do that.

Got it, I'm on it.

> However, we will still need a way to override a particular attribute/class in a lutaml-model class tree.

For attributes, I'm thinking of keeping a hash because attributes will always be strings. So we can update the value of a single attribute for the element by changing the hash value. 

For overriding the class I think we can provide methods to update attributes so we can change their type so it can be handled with different class. i.e `Person.update_attribute(:name, { type: MyCustomStringClass })`. Something like this.

@ronaldtse What do you think?

**Comment by ronaldtse** [comment]
Status on the two asks:

- **Passthrough (raw string)**: available today via `map_all to: :attr` (verbatim subtree) or `raw: true` on an element mapping; what's missing for the oasis-etm `entry` case is exactly #223's coexistence semantics (raw capture of *unclaimed* content alongside mapped siblings).
- **Swap the attribute's class**: achievable today by subclassing the model and overriding the attribute declaration (`class SpecialEntry < Entry; attribute :content, SpecialContent; end`) — the #295 tree-derivation mechanism would make this declarative (`override:` in import_model).

Both halves are now scoped under #223 (raw coexistence) and #295 (tree derivation). Keeping this open as the umbrella for the combined mechanism; comments welcome on the DSL shape.

**Comment by HassanAkbar** [comment]
Agreed, keeping this open as the umbrella until both halves land in #223 and #295.

**Comment by ronaldtse** [comment]
The passthrough half is done: #223 (PR #807) lets a mapped class keep structured attributes while `map_all` captures the remaining raw content — exactly the oasis-etm `<entry>` accept-anything case. The swap-out half (declarative tree derivation with `override:`) remains scoped in #295; keeping this closed for the passthrough mechanism — follow #295 for the swap-out design.
